internal/controller/clusterdeployment_reconciler_test.go

```diff
@@ -330,7 +332,183 @@ var _ = Describe("Reconcile", func() {
 					},
 				},
 			},
+			Spec: hivev1.ClusterDeploymentSpec{
+				Installed: true,
+			},
+			Status: hivev1.ClusterDeploymentStatus{
+				Conditions: []hivev1.ClusterDeploymentCondition{
+					{
+						Type:    hivev1.ClusterInstallRequirementsMetClusterDeploymentCondition,
+						Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+						Reason:  "ClusterInstallationStopped",
+						Message: "The cluster installation stopped",
+					},
+					{
+						Type:    hivev1.ClusterInstallStoppedClusterDeploymentCondition,
+						Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+						Reason:  "ClusterInstallStopped",
+						Message: "The installation has stopped because it completed successfully",
+					},
+					{
+						Type:    hivev1.ClusterInstallCompletedClusterDeploymentCondition,
+						Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+						Reason:  "InstallationCompleted",
+						Message: "The installation has completed: Cluster is installed",
+					},
+					{
+						Type:    hivev1.ClusterInstallFailedClusterDeploymentCondition,
+						Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+						Reason:  "InstallationNotFailed",
+						Message: "The installation has not failed",
+					},
+				},
+			},
 		}
+
+		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
+
+		res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: key})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		ci := &v1alpha1.ClusterInstance{}
+		Expect(c.Get(ctx, key, ci)).To(Succeed())
+
+		found := false
+		for i := range ci.Status.Conditions {
+			if ci.Status.Conditions[i].Type == string(conditions.Provisioned) &&
+				ci.Status.Conditions[i].Status == metav1.ConditionTrue &&
+				ci.Status.Conditions[i].Reason == string(conditions.Completed) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue())
```

Comment:
Might it be easier here to find the condition by type then assert on the status and reason?
I think it would also make the error message (if this were to fail) more clear.

Reply:
This applies to all the cases of this kind of condition checking (I think there's 3 in this PR).

Reply:
Nice suggestion!